libcontainer/process_linux.go

```diff
@@ -561,6 +561,13 @@ func (p *initProcess) start() (retErr error) {
 		}
 	}()
 
+	// A SIGKILL can happen at any time, and without the state.json, 
+	// the 'runc delete --force' command won't be able to clear the cgroup.
+	_, err = p.container.updateState(p)
```

Comment:
Thanks.
We should know that we will write the pid of init process to `state.json`, so when we do `delete -f`, once this init process has dead, but runc stage 2 process is alive, I think maybe we still can't remove this cgroup path either because there is still one process in this cgroup.

Reply:
In my testing, I discovered that when systemd is used to handle cgroup, it gets cleaned up once runc init has exited. However, if we don't use systemd, some remnants of cgroup are left behind.

If we're not using systemd to manage, we might need to do a check in runc delete to see if runc init is still listed in cgroups.procs. This shouldn't take too much time since runc init is bound to exit due to an error. This error happens because when runc init gets to parts like procHooks that require synchronization with the parent process, it fails as the parent process has already been terminated, leading to errors when runc init tries to write or read values from the pipeline, and consequently, it exits.

<img width="979" alt="{BF5FEB57-6023-4623-A04F-7C96FC189661}" src="https://github.com/user-attachments/assets/e1df048a-642d-450e-8f71-7103a105ae2b">

<img width="924" alt="{F284436E-31E8-4967-AE89-18A124B1C24C}" src="https://github.com/user-attachments/assets/941e8f04-ccde-4e82-980e-2aabe9794599">